nelox/interpreter.py

```diff
@@ -0,0 +1,223 @@
+from nelox.Expr import Literal, Variable, List
+
+
+class Environment:
+    def __init__(self, parent=None):
+        self.values = {}
+        self.parent = parent
+
+    def get(self, name):
+        if name in self.values:
+            return self.values[name]
+        elif self.parent:
+            return self.parent.get(name)
+        else:
+            raise RuntimeError(f"Undefined variable '{name}'")
+
+    def set(self, name, value):
+        if name in self.values:
+            self.values[name] = value
+        elif self.parent:
+            try:
+                self.parent.set(name, value)
+            except RuntimeError:
+                self.values[name] = value
+        else:
+            self.values[name] = value
+
+
+def _add(*args):
+    if not args:
+        raise RuntimeError("'+' requires at least one argument")
+
+    result = args[0]
+    for arg in args[1:]:
+
+        if isinstance(result, list) and isinstance(arg, (int, float)):
+            result = [x + arg for x in result]
+        elif isinstance(result, (int, float)) and isinstance(arg, list):
+            result = [result + x for x in arg]
+
+        elif isinstance(result, (int, float)) and isinstance(arg, (int, float)):
+            result = result + arg
+
+        else:
+            result = str(result) + str(arg)
```

Comment:
It is a bit strange that integers work pointwise but strings have to be treated individually.  Why not let `(+ (list "Hello" "world") "!")` work?  It seems like the logic should be similar.